### PR TITLE
code-gen: add +groupGoName to allow unique Go identifiers in clientsets and informers

### PR DESCRIPTION
**Pull request by sttts**
We use the first segement of the GroupName as Go name. Hence, a GroupName
"policy.k8s.io" and "policy.authorization.k8s.io" could not live in the
same clientset or shared informer factory. This PR add another tag:

```
  // +groupGoName=AuthorizationPolicy
```

### CODE REVIEW

File: staging/src/k8s.io/code-generator/cmd/client-gen/generators/client_generator.go [hunk 1]
```diff
@@ -310,9 +316,18 @@ func Packages(context *generator.Context, arguments *args.GeneratorArgs) generat
 	applyGroupOverrides(context.Universe, customArgs)
 
 	gvToTypes := map[clientgentypes.GroupVersion][]*types.Type{}
+	groupGoNames := make(map[clientgentypes.GroupVersion]string)
 	for gv, inputDir := range customArgs.GroupVersionToInputPath {
-		// Package are indexed with the vendor prefix stripped
 		p := context.Universe.Package(path.Vendorless(inputDir))
+
+		// If there's a comment of the form "// +groupGoName=SomeUniqueShortName", use that as
+		// the Go group identifier in CamelCase. It defaults
```
Comment (mfojtik) [inline staging/src/k8s.io/code-generator/cmd/client-gen/generators/client_generator.go]: It defaults to ?

**Comment by sttts** [comment]
/assign @deads2k 

For approval.

**Pull request merged**